apps/api/src/app/models/lessons/lessons.service.ts

```diff
@@ -125,6 +185,26 @@ export class LessonsModelService {
           ],
           as: "categories"
         }
+      },
```

Comment:
you duplicating same query many times, you can reuse it, double check all your query.